frontend/src/store/modules/sheet.ts

```diff
@@ -259,7 +262,12 @@ export const useSheetStore = defineStore("sheet", {
     },
     async getOrFetchSheetById(sheetId: SheetId) {
       const storedSheet = this.sheetById.get(sheetId);
-      if (storedSheet && storedSheet.id !== UNKNOWN_ID) {
+      if (
+        storedSheet &&
+        storedSheet.id !== UNKNOWN_ID &&
+        // if sheet size is equal to  INCOMPLETE_SHEET_SIZE, we still think it's incomplete.
```

Comment:
What if the sheet is exactly 10240 bytes? Though we can leave a TODO and address it later.